src/services/Hooks/useCheckPermissions.ts

```diff
@@ -0,0 +1,41 @@
+import * as Notifications from 'expo-notifications';
+import * as Permissions from 'expo-permissions';
+import axios from 'axios';
+import { secretIp } from '../../../secrets/secrets';
+import { AsyncStorage } from 'react-native';
+import { extractId } from '../../store/helper';
+
+export const useCheckPermissions = async () => {
+    let accesskey = await AsyncStorage.getItem('access_token');
+    const userId = extractId(accesskey);
+
+    try {
+        let pushToken;
+        let permissionStatus = await Permissions.getAsync(Permissions.NOTIFICATIONS);
+        if (permissionStatus.status !== 'granted') {
+            permissionStatus = await Permissions.askAsync(Permissions.NOTIFICATIONS);
```

Comment:
what's the difference between getAsync and askAsync

Reply:
getAsync - retrieving stuff asynchronously from permissions api
Permissions.askAsync - displays the "allow this app to send notifications" thingy and they can select yes/no